test/unit/test.sign.js

```diff
@@ -894,6 +898,48 @@ describe('sign', function() {
     }).catch(done);
   });
 
+  it('removes created XPI after signing', function(done) {
+    var unsignedXpiPath = '/tmp/some/non-existant/file.xpi';
+    var mockManifestGetter = new CallableMock({
+      returnValue: when.promise(function(resolve) {
+        resolve({});  // resolve with an empty manifest.
+      }),
+    });
+    var mockXPICreator = new CallableMock({
+      returnValue: when.promise(function(resolve) {
+        resolve(unsignedXpiPath);
+      }),
+    });
+    runSignCmd({
+      getManifest: mockManifestGetter.getCallable(),
+      createXPI: mockXPICreator.getCallable(),
+    }).then(function() {
+      expect(fakeFileRemover.call[0]).to.be.equal(unsignedXpiPath);
+      expect(mockProcessExit.call[0]).to.be.equal(0);
+      done();
+    }).catch(done);
+  });
+
+  it('does not remove custom XPI after signing', function(done) {
```

Comment:
because it's a custom XPI, which is part of the test name. I can change it to external.